### PR TITLE
fix profit calculation to exclude new deposits

### PR DESCRIPTION
**Pull request by coolhill**
- [x] in harvest, store `totalInvested` after `_rebalancePosition()` (since it might change during it)
- [x] added unit tests to check for correct behavior
- [x] decrease `totalInvested` on `_withdrawToVault()` (otherwise it doubles on `withdrawToVault()+depositIntoStrategy()`)
- [x] withdraw extra collateral when deleveraging to cover slippage losses. unrealized slippage ends up in vault

### CODE REVIEW

**Review by dxganta — APPROVED** [review]

**Review by brkomir — COMMENTED** [review]
Looks fine but I think we need to add a test to confirm "harvest"/"_rebalancePosition" works in the case when profits are negative

File: src/steth/scWETH.sol [hunk 1]
```diff
@@ -360,6 +360,9 @@ contract scWETH is sc4626, IFlashLoanRecipient {
         // needed otherwise counted as profit during harvest
         totalInvested += amount;
 
+        // when deleveraging, withdraw extra to cover slippage
+        if (!isDeposit) amount = flashLoanAmount.mulWadDown(C.ONE - slippageTolerance);
```
Comment (coolhill) [inline src/steth/scWETH.sol:364]:
```suggestion
        if (!isDeposit) amount += flashLoanAmount.mulWadDown(C.ONE - slippageTolerance);
```

**Review by brkomir — APPROVED** [review]
